nncf/tensor/functions/numpy_numeric.py

```diff
@@ -217,7 +217,7 @@ def _(
     axis: Optional[Union[int, Tuple[int]]] = None,
     keepdims: bool = False,
 ) -> Union[np.ndarray, np.generic]:
-    return np.array(np.quantile(a, q=q, axis=axis, keepdims=keepdims))
+    return np.array(np.quantile(a.astype(np.float64, copy=False), q=q, axis=axis, keepdims=keepdims))
```

Comment:
To keep behavior of quantile function changed input type of quantile function.
In numpy2 calculation of float type number execute according type of input array. 

| input dtype | 1.26| 2.0 |
| - | - | -|
|    np.int8 | float64| float64|
|    np.int32 | float64| float64|
|    np.int16 | float64| float64|
|    np.float16 | float64| **float16**|
|    np.float32 | float64| **float32**|
|    np.float64 | float64| **float64**|